rfcs/127-add-serverinternel-and-requestthrottled-errors.md

```diff
@@ -0,0 +1,41 @@
+- Author: xxchan <xxchan22f@gmail.com>
+- Start Date: 2021-06-24
+- RFC PR: [beyondstorage/specs#127](https://github.com/beyondstorage/specs/issues/127)
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
+
+# GSP-127: Add ServiceInternel and RequestThrottled Errors
+
+## Background
+
+These are common errors, especially in HTTP-based services, but also apply to RPC-based services. And they are retryable errors. If we provide them, users can retry only them instead of all errors.
+
+## Proposal
+
+I propose to add the following global error codes:
+- `ErrServerInternal`: e.g., HTTP 5xx
+- `ErrRequestThrottled`: e.g., HTTP 429 Too Many Requests/Limit Exceeded, 503 SlowDown, ...
+
+## Rationale
+
+Alternative names
+- `ErrServer`/`ErrInternal` & `ErrThrottling`/`ErrThrottled`: short, but not conform to `Err<Noun><Predicate>`
+- `ErrServerThrottling`: maybe not as common as `RequestThrottled`?
```

Comment:
How about `ServiceRequestThrottled`? So that we can make the scope more clear?

Reply:
I think "service request" seems strange

Reply:
OK, `RequestThrottled` looks good to me.